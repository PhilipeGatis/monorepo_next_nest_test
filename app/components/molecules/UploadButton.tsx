import React, { ChangeEvent, useState } from 'react'
import Button from '@mui/lab/LoadingButton'
import Snackbar from '@mui/material/Snackbar'
import IconButton from '@mui/material/IconButton'
import CloseIcon from '@mui/icons-material/Close'
import { uploadProductsList } from '../../lib/api'

const UploadButton: React.FC = () => {
  const [isUploading, setIsUploading] = useState<boolean>(false)
  const [message, setMessage] = useState<string>('')

  const upload = async (event: ChangeEvent<HTMLInputElement>) => {
    try {
      await uploadProductsList(event.target.files[0], event => {
        setIsUploading(true)
      })
      setMessage('Arquivo enviado com sucesso.')
    } catch (error) {
      if (error.signal === 422) {
        setMessage('Tipo de aquivo n√£o suportado.')
        return
      }
      setMessage('Error interno do sistema por favor tente novamente em alguns instantes.')
    } finally {
      setIsUploading(false)
    }
  }

  return (
    <>
      <input type="file" onChange={upload} accept="text/plain" style={{ display: 'none' }} id="contained-button-file" />
      <label htmlFor="contained-button-file">
        <Button loading={isUploading} variant="contained" color="primary" component="span">
          Upload
        </Button>
      </label>
      <Snackbar
        open={!!message}
        autoHideDuration={6000}
        onClose={() => setMessage('')}
        message={message}
        action={
          <>
            <IconButton size="small" aria-label="close" color="inherit" onClick={() => setMessage('')}>
              <CloseIcon fontSize="small" />
            </IconButton>
          </>
        }
      />
    </>
  )
}

export default UploadButton
