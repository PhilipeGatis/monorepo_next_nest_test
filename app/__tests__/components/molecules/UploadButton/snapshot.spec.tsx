import { render } from '@testing-library/react'
import UploadButton from '../../../../components/molecules/UploadButton'

describe('UploadButton', () => {
  it('render', () => {
    const { container } = render(<UploadButton />)
    expect(container).toMatchSnapshot()
  })
})
