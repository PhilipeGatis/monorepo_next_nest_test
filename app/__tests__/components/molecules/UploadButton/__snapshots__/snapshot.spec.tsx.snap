// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton render 1`] = `
<div>
  <input
    accept="text/plain"
    id="contained-button-file"
    style="display: none;"
    type="file"
  />
  <label
    for="contained-button-file"
  >
    <span
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r0:"
      role="button"
      tabindex="0"
    >
      Upload
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
  </label>
</div>
`;
